pkg/snapshot/velero.go

```diff
@@ -0,0 +1,74 @@
+package snapshot
+
+import (
+	"github.com/replicatedhq/kots/pkg/kotsadm"
+
+	"context"
+
+	"github.com/pkg/errors"
+	veleroclientv1 "github.com/vmware-tanzu/velero/pkg/generated/clientset/versioned/typed/velero/v1"
+	kuberneteserrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+func CreateRBACPermissions(kotsadmNamespace string) error {
+	veleroNamespace, err := DetectVeleroNamespace()
+	if err != nil {
+		return errors.Wrap(err, "failed to detect velero namespace")
+	}
+	if veleroNamespace == "" {
+		return errors.New("velero not found")
+	}
+
+	cfg, err := config.GetConfig()
+	if err != nil {
+		return errors.Wrap(err, "failed to get cluster config")
+	}
+
+	clientset, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return errors.Wrap(err, "failed to create kubernetes clientset")
+	}
+
+	if err := kotsadm.EnsureKotsadmRole(veleroNamespace, clientset); err != nil {
+		return errors.Wrap(err, "failed to ensure kotsadm role")
+	}
+
+	if err := kotsadm.EnsureKotsadmRoleBinding(veleroNamespace, kotsadmNamespace, clientset); err != nil {
+		return errors.Wrap(err, "failed to ensure kotsadm rolebinding")
+	}
+
+	return nil
+}
+
+func DetectVeleroNamespace() (string, error) {
```

Comment:
This looks like a function that's already implemented elsewhere.

Reply:
yes, it is already defined in `kotsadm`, but it's already part of the refactor in my other NFS pr, and didn't want to refactor kotsadm again to reduce conflicts.